scripts/nagios/check-rabbitmq-queue

```diff
@@ -31,50 +36,147 @@ states = {
     3: "UNKNOWN"
 }
 
-pattern = re.compile(r'(\w+)\t(\d+)')
-output = subprocess.check_output(['/usr/sbin/rabbitmqctl', 'list_queues'], universal_newlines=True)
+MAX_SECONDS_TO_CLEAR_FOR_BURSTS = 600
+MAX_SECONDS_TO_CLEAR_NORMAL = 120
+CRITICAL_SECONDS_TO_CLEAR_FOR_BURSTS = 1200
+CRITICAL_SECONDS_TO_CLEAR_NORMAL = 240
+
+def analyze_queue_stats(queue_name: str, stats: Dict[str, Any]) -> Dict[str, Any]:
+    now = int(time.time())
+    if stats == {}:
+        return dict(status=3,
+                    name=queue_name,
+                    message='invalid or no stats data')
+
+    if now - stats['update_time'] > 300 and queue_counts_rabbitmqctl[queue_name] > 10:
+        # Queue isn't updating the stats file and has some events in the backlog, it's
+        # likely stuck.
+        return dict(status=2,
+                    name=queue_name,
+                    message='queue appears to be stuck, last update %s' % (stats['update_time'],))
```

Comment:
Do we want to have the case where now isn't too old and the queue is <= 10 have an else clause?

Might be nice to not have to think about a month-old stats file in the logic below.

Reply:
https://github.com/zulip/zulip/pull/14240#discussion_r398717551 addresses this - the queue checker only checks active queues, and queue workers on startup refresh the file, exactly to avoid this situation.